### DISCUSSION TITLE
How to subscribe to mqtt device trigger

### DISCUSSION THREAD

**Discussion by grisia** [net-daemon/netdaemon#563] [General]
Hello, 

I'm trying to integrate my Homematic devices into the Home Assistant.
Is there a way to subscribe to a 'device trigger' event?

Here is the discovery payload I used. 

```bash
mosquitto_pub  -h 127.0.0.1 -u mosquitto -t "homeassistant/device_automation/remote1/button4/config" -m '{"unique_id":"738c1d8e-e3e4-4918-83bf-c1fbda8a5de2","automation_type":"trigger","type":"button_short_press","subtype":"button_4","payload":"click","topic":"homematic/remote1/button4","device":{"identifiers":["PEQ0781546"],"name":"Remote1","model":"HM-RC-4-3","manufacturer":"eQ-3 AG"}}'
```

I am currently able to trigger such automation in Home Assistant.

In NetDaemon I can then see that this automation has been triggered. But now I would not like to create a dummy automation for all mqtt triggers.

Is there a way to subscribe to mqtt trigger?

**Comment by grisia** [comment]
Maybe I can answer that myself: https://github.com/AppDaemon/appdaemon/issues/958

I think for the hand knitted homematic mqtt entities, it would be best if I used a binary sensor instead of a device trigger.
I just think I will face the same problem the next time I try to integrate my Zigbee devices via https://www.zigbee2mqtt.io/?

**Reply by helto4real** [reply]
Aight! I think there are some users that got MQTT access to work directly into an app. Find us in discord group and ask there. We are planning to support devices as well in the future. For now only entity id:s can be used subscribing to state changes.
